File: PokeLangPython/Lexer/Number.py
from AFD import AFD
from Token import Token

class Number(AFD):
    def evaluate(self, code, linha, column):
        number = ""

        if code.isdigit():
            number = self.read_number(code)

            if self.end_number(code):
                return Token("NUM", number, linha, column)

            if code.startswith('.'):
                number += '.'
                code = next(code, None)
                number += self.read_number(code)

                if self.end_number(code):
                    return Token("FLUTUANTE", number, linha, column)

        return None

    def read_number(self, code):
        number = ""

        while code.isdigit():
            number += code
            code = next(code, None)

        return number

    def end_number(self, code):
        return code in (' ', '+', '-', '*', '(', ')', '{', '}', '[', ']', ';', '\n', None)
    
    def skip_white_space(self, code):  # Adiciona esta função
        while code.current() in (' ', '\n', '\r', '\t'):
            code.next()

File: PokeLangJava/src/AFD/Number.java
package AFD;

import java.text.CharacterIterator;
import Token.Token;

public class Number extends AFD {
	@Override
	public Token evaluate(CharacterIterator code) {
		String number;

		if (Character.isDigit(code.current())) {
			number = readNumber(code);

			if (endNumber(code)) {
				return new Token("NUM", number);
			}

			if (code.current() == '.') {
				number += '.';
				code.next();
				number += readNumber(code);

				if (endNumber(code)) {
					return new Token("FLUTUANTE", number);
				}
			}
		}

		return null;
	}

	private String readNumber(CharacterIterator code) {
		String number = "";

		while (Character.isDigit(code.current())) {
			number += code.current();
			code.next();
		}

		return number;
	}

	private boolean endNumber(CharacterIterator code) {
		return code.current() == ' '
				|| code.current() == '+'
				|| code.current() == '-'
				|| code.current() == '*'
				|| code.current() == '('
				|| code.current() == ')'
				|| code.current() == '{'
				|| code.current() == '}'
				|| code.current() == '['
				|| code.current() == ']'
				|| code.current() == ';'
				|| code.current() == '\n'
				|| code.current() == CharacterIterator.DONE;
	}
}
